File: Algorithms/python/070-ClimbingStairs/ClimbingStairs.py
class Solution(object):
    def climbStairs(self, n):
        """
        :type n: int
        :rtype: int
        """
        steps = [0] * (n + 1)

        for i in range(n + 1):
            if i == 0:
                steps[i] = 1
                continue
            if i >= 1:
                steps[i] += steps[i - 1]
            if i >= 2:
                steps[i] += steps[i - 2]

        return steps[n]

File: Algorithms/cpp/ClimbingStairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        int d[n + 1];
        d[0] = 1;
        for (int i = 1; i <= n; i++) {
            d[i] = d[i - 1];
            d[i] += i > 1 ? d[i - 2] : 0;
        }
        return d[n];
    }
};
